tree/ntuple/v7/src/RField.cxx

```diff
@@ -782,12 +782,14 @@ ROOT::Experimental::RClassField::RClassField(std::string_view fieldName, std::st
       throw RException(R__FAIL(std::string(className) + " is not supported"));
    }
 
+   int i = 0;
    for (auto baseClass : ROOT::Detail::TRangeStaticCast<TBaseClass>(*fClass->GetListOfBases())) {
       TClass *c = baseClass->GetClassPointer();
-      auto subField = Detail::RFieldBase::Create(std::string(kPrefixInherited) + c->GetName(),
+      auto subField = Detail::RFieldBase::Create(std::string(kPrefixInherited) + "_" + std::to_string(i),
```

Comment:
This is highly dependent on the order in which `TClass` enumerates the base classes, which in turn depends on the inheritance graph (?).

As-is, `std::variant<...>` is also dependent on the order, but I think this is what we are looking for in both cases.